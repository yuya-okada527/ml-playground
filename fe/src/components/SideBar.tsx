import {
  createStyles,
  Grid,
  List,
  ListItem,
  makeStyles,
  Theme,
  Typography,
} from "@material-ui/core";
import HomeIcon from "@material-ui/icons/Home";
import InfoIcon from "@material-ui/icons/Info";
import Link from "next/link";
import React from "react";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      height: "calc(100vh - 170px)",
    },
    sideBarItem: {
      color: theme.palette.primary.main,
      cursor: "pointer",
      borderRadius: "5px",
      "&:hover": {
        backgroundColor: "#eee",
      },
    },
    sidebarIcon: {
      margin: theme.spacing(1),
    },
    sidebarText: {
      marginTop: theme.spacing(0.5),
    },
  })
);

/**
 * サイドバーを描画する
 */
const SideBar: React.FC = () => {
  const classes = useStyles();
  return (
    <List className={classes.root}>
      <ListItem>
        <Link href="/">
          <Grid container className={classes.sideBarItem}>
            <HomeIcon className={classes.sidebarIcon} />
            <Typography className={classes.sidebarText} variant="h6">
              Home
            </Typography>
          </Grid>
        </Link>
      </ListItem>
      <ListItem>
        <Link href="/about">
          <Grid container className={classes.sideBarItem}>
            <InfoIcon className={classes.sidebarIcon} />
            <Typography className={classes.sidebarText} variant="h6">
              About
            </Typography>
          </Grid>
        </Link>
      </ListItem>
    </List>
  );
};

export default SideBar;
