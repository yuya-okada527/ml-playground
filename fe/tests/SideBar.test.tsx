import React from "react";
import { shallow } from "enzyme";
import SideBar from "../src/components/SideBar";

describe("SideBar", () => {
  it("Shapshot Test", () => {
    const sideBar = shallow(<SideBar />);
    expect(sideBar).toMatchSnapshot();
  });
});
