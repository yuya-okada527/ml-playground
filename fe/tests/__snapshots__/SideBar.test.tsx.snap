// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar Shapshot Test 1`] = `
<WithStyles(ForwardRef(List))
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(ListItem))>
    <Link
      href="/"
    >
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-sideBarItem-2"
        container={true}
      >
        <Memo(ForwardRef(HomeIcon))
          className="makeStyles-sidebarIcon-3"
        />
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-sidebarText-4"
          variant="h6"
        >
          Home
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </Link>
  </WithStyles(ForwardRef(ListItem))>
  <WithStyles(ForwardRef(ListItem))>
    <Link
      href="/about"
    >
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-sideBarItem-2"
        container={true}
      >
        <Memo(ForwardRef(InfoIcon))
          className="makeStyles-sidebarIcon-3"
        />
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-sidebarText-4"
          variant="h6"
        >
          About
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </Link>
  </WithStyles(ForwardRef(ListItem))>
</WithStyles(ForwardRef(List))>
`;
